src/tdycore.c

```diff
@@ -276,7 +276,7 @@ PetscErrorCode TDyCreateGrid(TDy tdy) {
   }
 
   if (!tdy->dm) {
-    DM dm;
+    DM dm = tdy->dm;
```

Comment:
I think you can leave this as just a declaration without the assignment, since it's set in the next line.